src/impl_methods.rs

```diff
@@ -1218,6 +1218,23 @@ where
         D::is_contiguous(&self.dim, &self.strides)
     }
 
+    pub fn as_standard_layout(&self) -> CowArray<'_, A, D>
+    where
+        S: Data<Elem = A>,
+        A: Clone,
+    {
+        if self.is_standard_layout() {
+            CowArray::from(self.view())
+        } else {
+            let v = self.iter().cloned().collect::<Vec<A>>();
+            let owned_array: Array<A, D> = unsafe {
+                // Safe because we use shape and content of existing array here.
+                ArrayBase::from_shape_vec_unchecked(self.raw_dim(), v)
```

Comment:
This makes me a bit uncomfortable, especially because I don't know if in the future it will be possible for users to specialize the implementation of `FromIterator<T> for Vec<T>`. Let's add an assertion that the length of `v` is correct:

```rust
let v: Vec<A> = self.iter().cloned().collect();
let dim = self.dim.clone();
assert_eq!(v.len(), dim.size());
let owned_array: Array<A, D> = unsafe {
    // Safe because the shape and element type are from the existing array
    // and the strides are the default strides.
    Array::from_shape_vec_unchecked(dim, v)
};
CowArray::from(owned_array)
```